lager/reader.hpp

```diff
@@ -42,6 +42,12 @@ struct reader_mixin
         return attred(member, *this);
     }
 
+    template <typename Xform>
+    auto xf(Xform&& xf)
+    {
+        return xform(xf)(*this);
```

Comment:
It seems you're missing a forward here with `xf` (also in the writer xf impl)

Reply:
Good catch!